nilearn/datasets/neurovault.py

```diff
@@ -0,0 +1,2485 @@
+    nilearn.datasets.neurovault.LessOrEqual,
+    nilearn.datasets.neurovault.LessThan,
+    nilearn.datasets.neurovault.IsIn,
+    nilearn.datasets.neurovault.NotIn,
+    nilearn.datasets.neurovault.Contains,
+    nilearn.datasets.neurovault.NotContains,
+    nilearn.datasets.neurovault.Pattern.
+
+    """
+    def __eq__(self, other):
+        return not bool(other)
+
+
+class NotNull(_SpecialValue):
+    """Special value used to filter terms.
+
+    An instance of this class will always be equal to, and only to,
+    any non-zero value of any type (by non-zero we mean for which bool
+    returns True).
+
+    See Also
+    --------
+    nilearn.datasets.neurovault.IsNull,
+    nilearn.datasets.neurovault.NotEqual,
+    nilearn.datasets.neurovault.GreaterOrEqual,
+    nilearn.datasets.neurovault.GreaterThan,
+    nilearn.datasets.neurovault.LessOrEqual,
+    nilearn.datasets.neurovault.LessThan,
+    nilearn.datasets.neurovault.IsIn,
+    nilearn.datasets.neurovault.NotIn,
+    nilearn.datasets.neurovault.Contains,
+    nilearn.datasets.neurovault.NotContains,
+    nilearn.datasets.neurovault.Pattern.
+
+    """
+    def __eq__(self, other):
+        return bool(other)
+
+
+class NotEqual(_SpecialValue):
+    """Special value used to filter terms.
+
+    An instance of this class is constructed with `NotEqual(obj)`. It
+    will allways be equal to, and only to, any value for which
+    ``obj == value`` is ``False``.
+
+    Parameters
+    ----------
+    negated : object
+        The object from which a candidate should be different in order
+        to pass through the filter.
+
+    See Also
+    --------
+    nilearn.datasets.neurovault.IsNull,
+    nilearn.datasets.neurovault.NotNull,
+    nilearn.datasets.neurovault.GreaterOrEqual,
+    nilearn.datasets.neurovault.GreaterThan,
+    nilearn.datasets.neurovault.LessOrEqual,
+    nilearn.datasets.neurovault.LessThan,
+    nilearn.datasets.neurovault.IsIn,
+    nilearn.datasets.neurovault.NotIn,
+    nilearn.datasets.neurovault.Contains,
+    nilearn.datasets.neurovault.NotContains,
+    nilearn.datasets.neurovault.Pattern.
+
+    """
+    def __init__(self, negated):
+        self.negated_ = negated
+        self.repr_arg_ = self.negated_
+
+    def __eq__(self, other):
+        return not self.negated_ == other
+
+
+class _OrderComp(_SpecialValue):
+    """Base class for special values based on order comparisons."""
+    def __init__(self, bound):
+        self.bound_ = bound
+        self._cast = type(bound)
+        self.repr_arg_ = self.bound_
+
+    def __eq__(self, other):
+        try:
+            return self._eq_impl(self._cast(other))
+        except (TypeError, ValueError):
+            return False
+
+
+class GreaterOrEqual(_OrderComp):
+    """Special value used to filter terms.
+
+    An instance of this class is constructed with `GreaterOrEqual(obj)`. It
+    will allways be equal to, and only to, any value for which
+    ``obj <= value`` is ``True``.
```

Comment:
this is used to specify that whatever we compare it to should be greater or equal to the value we passed to the constructor. Therefore if we pass obj to the constructor, when we compare it to value, we get true if value is greater or equal to obj